components/task/details/solutions/newSolutionForm/NewSolutionForm.tsx

```diff
@@ -0,0 +1,180 @@
+import React from 'react';
+import Button, { ButtonLayout } from '../../../../shared/button/Button';
+import Select from 'react-select';
+import { formStyles } from './formStyles';
+import clsx from 'clsx';
+import NewSolutionFormInput from './newSolutionFormInput/NewSolutionFormInput';
+import { Form, Field } from 'react-final-form';
+import { FieldValidator } from 'final-form';
+import { URLValidator } from '../../../../../utils/validators/URLValidator';
+import { required } from '../../../../../utils/validators/requiredValidator';
+
+export const technologies = [
+  { value: 'html', label: '#html' },
+  { value: 'css', label: '#css' },
+  { value: 'js', label: '#js' },
+  { value: 'nodejs', label: '#nodejs' },
+  { value: 'python', label: '#python' },
+  { value: 'react', label: '#react' },
+  { value: 'vue', label: '#vue' },
+  { value: 'angular', label: '#angular' },
+];
+
+interface Technology {
+  value: string;
+  label: string;
+}
+
+interface NewSolutionFormProps {
+  setIsModalOpen: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+interface TechnologiesSelect {
+  label: string;
+  value: string;
+}
+interface FormValues {
+  solutionLinkInput: string;
+  liveLinkInput?: string;
+  technologiesSelect: TechnologiesSelect[];
+  reviewCheckbox?: boolean;
+}
+
+const NewSolutionForm: React.FC<NewSolutionFormProps> = ({ setIsModalOpen }) => {
+  function sendSolutions(values: FormValues): FormValues {
+    return values;
+  }
+
+  const onSubmit = async (values: FormValues): Promise<void> => {
+    try {
+      const data = await sendSolutions(values);
+      console.log(JSON.stringify(data));
+    } catch (error) {
+      return error.message;
+    }
+  };
+
+  function composeValidators(...validators: FieldValidator<string>[]): FieldValidator<string> {
```

Comment:
to do osobnego pliku gdzies w okolicach utils/validation

Reply:
zrobione